src/Site.js

```diff
@@ -131,7 +131,9 @@ class Site {
     this.plugins = {};
     this.siteConfig = {};
     this.siteConfigPath = siteConfigPath;
-    this.userDefinedVariablesMap = {};
+
+    // Site wide variable preprocessor
+    this.variablePreprocessor = undefined;
```

Comment:
`userDefinedVariablesMap` is removed - note all variable preprocessing will be centered around `variablePreprocessor`